packages/go_router/lib/src/router.dart

```diff
@@ -509,6 +521,11 @@ class GoRouter implements RouterConfig<RouteMatchList> {
   String _effectiveInitialLocation(String? initialLocation) {
     final String platformDefault =
         WidgetsBinding.instance.platformDispatcher.defaultRouteName;
+    if (overridePlatformDefaultLocation) {
```

Comment:
should this be move to beginning of the method?

Reply:
cant as the underlying if conditions depend on `platformDefault`

Reply:
I am not sure if I understand, the if condition returns initialLocation directly, it doesn't use platformDefault

Reply:
`platformDefault` would send out a string:
 - `'/'`                    if no route is provided by platform on startup (eg. by deeplink)
 - `'/some-route'` if specific route is provided by platform on startup (eg. by deeplink)
 
 if we're using `overridePlatformDefaultLocation`, we don't care about the `platformDefault` in the first place and return `initialLocation` set by router.

Feel free to crosscheck
<details>
<summary> image </summary>

![image](https://github.com/flutter/packages/assets/84124091/9c982f71-1c3b-40d0-b29a-05dca6c5af5a)
</details>